test/vegeta/pacers/steady_up_pacer.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pacers
+
+import (
+	"fmt"
+	"math"
+	"time"
+
+	vegeta "github.com/tsenart/vegeta/lib"
+)
+
+// steadyUpPacer is a Pacer that describes attack request rates that increases in the beginning then becomes steady.
+//  Max  |     ,----------------
+//       |    /
+//       |   /
+//       |  /
+//       | /
+//  Min -+------------------------------> t
+//       |<-Up->|
+type steadyUpPacer struct {
+	// upDuration is the duration that attack request rates increase from Min to Max, must be larger than 0.
+	upDuration time.Duration
+	// min is the attack request rates from the beginning, must be larger than 0.
+	min vegeta.Rate
+	// max is the maximum and final steady attack request rates, must be larger than Min.
+	max vegeta.Rate
+
+	slope        float64
+	minHitsPerNs float64
+	maxHitsPerNs float64
+}
+
+// NewSteadyUpPacer returns a new SteadyUpPacer with the given config.
+func NewSteadyUpPacer(min vegeta.Rate, max vegeta.Rate, upDuration time.Duration) vegeta.Pacer {
+	if upDuration <= 0 || min.Freq <= 0 || min.Per <= 0 || max.Freq <= 0 || max.Per <= 0 {
+		panic("Configuration for this SteadyUpPacer is invalid!")
+	}
+	minHitsPerNs := hitsPerNs(min)
+	maxHitsPerNs := hitsPerNs(max)
+	if minHitsPerNs >= maxHitsPerNs {
+		panic("min rate must be smaller than max rate for SteadyUpPacer!")
```

Comment:
why not return an error?

Reply:
Done. Returned the error and added unit tests for it.